fvm/evm/emulator/state/base.go

```diff
@@ -0,0 +1,472 @@
+func (v *BaseView) GetNonce(addr gethCommon.Address) (uint64, error) {
+	acc, err := v.getAccount(addr)
+	nonce := uint64(0)
+	if acc != nil {
+		nonce = acc.Nonce
+	}
+	return nonce, err
+}
+
+// GetCode returns the code of an address
+//
+// for non-existent accounts or accounts without a code (e.g. EOAs) it returns nil
+func (v *BaseView) GetCode(addr gethCommon.Address) ([]byte, error) {
+	return v.getCode(addr)
+}
+
+// GetCodeHash returns the code hash of an address
+//
+// for non-existent accounts or accounts without a code (e.g. EOAs) it returns default empty
+// hash value (gethTypes.EmptyCodeHash)
+func (v *BaseView) GetCodeHash(addr gethCommon.Address) (gethCommon.Hash, error) {
+	acc, err := v.getAccount(addr)
+	codeHash := gethTypes.EmptyCodeHash
+	if acc != nil {
+		codeHash = acc.CodeHash
+	}
+	return codeHash, err
+}
+
+// GetCodeSize returns the code size of an address
+//
+// for non-existent accounts or accounts without a code (e.g. EOAs) it returns zero
+func (v *BaseView) GetCodeSize(addr gethCommon.Address) (int, error) {
+	code, err := v.GetCode(addr)
+	return len(code), err
+}
+
+// GetState returns values for an slot in the main storage
+//
+// for non-existent slots it returns the default empty hash value (gethTypes.EmptyCodeHash)
+func (v *BaseView) GetState(sk types.SlotAddress) (gethCommon.Hash, error) {
+	return v.getSlot(sk)
+}
+
+// UpdateSlot updates the value for a slot
+func (v *BaseView) UpdateSlot(sk types.SlotAddress, value gethCommon.Hash) error {
+	return v.storeSlot(sk, value)
+}
+
+// GetRefund returns the total amount of (gas) refund
+//
+// this method returns the value of zero
+func (v *BaseView) GetRefund() uint64 {
+	return 0
+}
+
+// GetTransientState returns values for an slot transient storage
+//
+// transient storage is not a functionality for the base view so it always
+// returns the default value for non-existent slots
+func (v *BaseView) GetTransientState(types.SlotAddress) gethCommon.Hash {
+	return gethCommon.Hash{}
+}
+
+// AddressInAccessList checks if an address is in the access list
+//
+// access list control is not a functionality of the base view
+// it always returns false
+func (v *BaseView) AddressInAccessList(gethCommon.Address) bool {
+	return false
+}
+
+// SlotInAccessList checks if a slot is in the access list
+//
+// access list control is not a functionality of the base view
+// it always returns false
+func (v *BaseView) SlotInAccessList(types.SlotAddress) (addressOk bool, slotOk bool) {
+	return false, false
+}
+
+// CreateAccount creates a new account
+func (v *BaseView) CreateAccount(
+	addr gethCommon.Address,
+	balance *big.Int,
+	nonce uint64,
+	code []byte,
+	codeHash gethCommon.Hash,
+) error {
+	var colID []byte
+	// if is an smart contract account
+	if len(code) > 0 {
+		err := v.storeCode(addr, code)
+		if err != nil {
+			return err
+		}
+
+		// create a new collection for slots
+		col, err := v.collectionProvider.NewCollection()
```

Comment:
Thats a good idea, given the recent changes it will be common to have accounts with code without any stored data. update the code logic at [47be287](https://github.com/onflow/flow-go/pull/5166/commits/47be2877c60cbccc53baa10d8fbc10a5fa30d313)